dspace-api/src/main/java/org/dspace/content/crosswalk/METSRightsCrosswalk.java

```diff
@@ -471,85 +471,31 @@ else if (element.getName().equals("Context"))
                         {
-                            rp.setRpName(element.getAttributeValue("rpName"));
-                        }
-                        try {
-                            if (element.getAttributeValue("start-date") != null)
-                            {
-                                rp.setStartDate(sdf.parse(element.getAttributeValue("start-date")));
-                            }
-                            if (element.getAttributeValue("end-date") != null)
-                            {
-                                rp.setEndDate(sdf.parse(element.getAttributeValue("end-date")));
-                            }
-                        }catch (ParseException ex) {
-                            java.util.logging.Logger.getLogger(METSRightsCrosswalk.class.getName()).log(Level.SEVERE, null, ex);
+                        	rp.setEndDate(sdf.parse(element.getAttributeValue("end-date")));
                         }
-
-                        List<Element> le = new ArrayList<Element>(element.getChildren());
-                        for (Element el : le)
-                        {
-                            if ((el.getAttributeValue("DISCOVER").equalsIgnoreCase("true")) 
-                                    && (el.getAttributeValue("DISPLAY").equalsIgnoreCase("true")))
-                            {
-                                if (el.getAttributeValue("DELETE").equalsIgnoreCase("false"))
-                                {
-                                    if (el.getAttributeValue("MODIFY").equalsIgnoreCase("false"))
-                                    {
-                                        rp.setAction(Constants.READ);
-                                    }
-                                    else
-                                    {
-                                        rp.setAction(Constants.WRITE);
-                                    }
-                                }
-                                else
-                                {
-                                    if (el.getAttributeValue("MODIFY").equalsIgnoreCase("true"))
-                                    {
-                                        rp.setAction(Constants.DELETE);
-                                        if ((el.getAttributeValue("COPY").equalsIgnoreCase("true"))
-                                                &&(el.getAttributeValue("DUPLICATE").equalsIgnoreCase("true"))
-                                                &&(el.getAttributeValue("PRINT").equalsIgnoreCase("true")))
-                                        {
-                                            rp.setAction(Constants.ADMIN);
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                        policies.add(rp);
-                    } catch (NullPointerException ex) {
```

Comment:
That `for` loop's permissions-decoding logic should be covered in the call to the `parsePermissions()` method (see the call on [line 595](https://github.com/Georgetown-University-Libraries/DSpace/blob/b387bca5e532b0d35fc6bdb31549410d6f6c2cb7/dspace-api/src/main/java/org/dspace/content/crosswalk/METSRightsCrosswalk.java#L595) and the method definition on [line 689)](https://github.com/Georgetown-University-Libraries/DSpace/blob/b387bca5e532b0d35fc6bdb31549410d6f6c2cb7/dspace-api/src/main/java/org/dspace/content/crosswalk/METSRightsCrosswalk.java#L689). I could not think of a scenario where a METS file context element would have more than one permissions child element (but folks more experienced with the internals, please speak up if I'm off-base here), and found that the looping logic would even sometimes cause an NPE when it looked at a non-permissions child element (see the [example context element](https://jira.duraspace.org/browse/DS-3140?focusedCommentId=49773&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-49773) provided in the corresponding JIRA ticket). So instead of looping through all of the child elements, I grabbed just the permissions child element (see [line 480](https://github.com/Georgetown-University-Libraries/DSpace/blob/b387bca5e532b0d35fc6bdb31549410d6f6c2cb7/dspace-api/src/main/java/org/dspace/content/crosswalk/METSRightsCrosswalk.java#L480)) and passed that to the `parsePermissions()` method.